conclave-host/src/main/kotlin/com/r3/conclave/host/EnclaveHost.kt

```diff
@@ -537,6 +424,18 @@ class EnclaveHost private constructor(
         if (hostStateManager.state is Started) return
         hostStateManager.checkStateIsNot<Closed> { "The host has been closed." }
 
+        // For now, there is no enclave for gramine instead the mock enclave is initialised.
+        // This is OK because no VM is started in Mock mode and this will allow us to integrate Gramine with Conclave
+        // iteratively without causing issues to the normal operation of Conclave.
+        // Start gramine only if the environment variable is set
+        if(Gramine.isGramineEnabled())
+        {
+            if(enclaveHandle !is MockEnclaveHandle) {
+                throw Exception("Gramine cannot be started in non-mock modes")
+            }
+            Gramine.start()
+        }
+
```

Comment:
If you have environmental variable set and you use non-mock mode, will it fail? Shouldn't it just ignore it in that case and maybe write a warning instead? Just an idea, not really fussy about the behaviour at this stage.

Reply:
I thought it would be better to have an explicit error for now to be easier to understand what is happening. I am happy to change the code if you prefer the log message.

Reply:
Let's leave it for now.